src/pages/SearchingPage/SearchingPage.tsx

```diff
@@ -38,19 +44,30 @@ export const SearchingPage = () => {
     setSearchParams({ user: newUser });
   }
 
-  const navigation = () => {
+  const onOpenRepo = () => {
     navigate("/repos", { state: { userData } });
   };
+
+  const isFound = (user: IUser | undefined) => {
+    if (userData?.hasOwnProperty("message")) {
+      return <h1> User not found !</h1>;
+    } else {
+      return <List userData={userData} onClickHandler={onOpenRepo} />;
+    }
+  };
+
+  const renderedComponent = isFound(userData);
+
   return (
     <main className={styles.mainWrapper}>
       <div className={styles.searchingPage}>
         <SearchInput
-          handleSubmit={handleSubmit}
+          handlerSubmit={handleSubmit}
           inputValue={userQuery}
           button={true}
           placeholder={"Search for Users"}
         />
-        {userData && <List userData={userData} onClickHandler={navigation} />}
+        {isLoading ? <Loader /> : userData && renderedComponent}
```

Comment:
const renderUserList = () => {
    if (userData?.hasOwnProperty("message")) {
      return <h1> User not found !</h1>;
    } else {
      return <List userData={userData} onClickHandler={onOpenRepo} />;
    }
  };

{isLoading ? <Loader /> : renderUserList()}